src/main/java/org/pesmypetcare/webservice/service/MedicationServiceImp.java

```diff
@@ -0,0 +1,54 @@
+package org.pesmypetcare.webservice.service;
+
+import org.pesmypetcare.webservice.dao.MedicationDao;
+import org.pesmypetcare.webservice.entity.MedicationEntity;
+import org.pesmypetcare.webservice.error.DatabaseAccessException;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.Map;
+
+public class MedicationServiceImp implements MedicationService {
+    @Autowired
+    private MedicationDao medicationDao;
+
+    @Override
+    public void createMedication(String owner, String petName, String dateName, MedicationEntity medication) {
+        medicationDao.createMedication(owner, petName, dateName, medication);
+    }
+
+    @Override
+    public void deleteByDateAndName(String owner, String petName, String dateName) {
+        medicationDao.deleteByDateAndName(owner, petName, dateName);
+    }
+
+    @Override
+    public void deleteAllMedications(String owner, String petName) throws DatabaseAccessException {
+        medicationDao.deleteAllMedications(owner, petName);
+    }
+
+    @Override
+    public MedicationEntity getMedicationData(String owner, String petName, String dateName) throws DatabaseAccessException {
+        return medicationDao.getMedicationData(owner, petName, dateName);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName) throws DatabaseAccessException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Line is longer than 120 characters (found 126).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)